Week3/app/src/main/java/com/florencenjeri/businesscard/MainActivity.kt

```diff
@@ -1,11 +1,88 @@
 package com.florencenjeri.businesscard
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.animation.AnimationUtils
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatDelegate
+import com.raywenderlich.airlock.Constants
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    lateinit var prefs: SharedPreferences
```

Comment:
I'd delegate this either by lazy, or fetch it from the App class!

Reply:
By fetching sharedPrefences from the app class do you mean instead of using
 ```kotlin
 prefs = this.getSharedPreferences(Constants.PREFS_MODE, Context.MODE_PRIVATE)
```  
I should instead use:
 ```kotlin  
 prefs = applicationContext.getSharedPreferences(Constants.PREFS_MODE, Context.MODE_PRIVATE)
```